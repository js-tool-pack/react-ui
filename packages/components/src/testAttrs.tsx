import { fireEvent, render } from '@testing-library/react';
import type { PropsBase } from '@pkg/shared';
import React from 'react';

export function testAttrs<
  T extends { attrs?: Partial<React.HTMLAttributes<HTMLElement>> },
>(component: React.FC<T>, displayName?: string) {
  const name = displayName || component.displayName || component.name;
  test(name ? `${name} attrs` : `attrs`, () => {
    const onClick = jest.fn();

    const FC = component as React.FC<PropsBase>;
    const { container } = render(
      <FC
        attrs={{
          style: { width: '20px' },
          className: 'foo',
          role: 'note',
          onClick,
        }}
      />,
    );

    expect(container.firstChild).toHaveClass('foo');
    expect(container.firstChild).toHaveAttribute('role', 'note');
    expect(container.firstChild).toHaveStyle({ width: '20px' });

    expect(onClick).not.toBeCalled();
    fireEvent.click(container.firstChild!);
    expect(onClick).toBeCalled();
  });
}
