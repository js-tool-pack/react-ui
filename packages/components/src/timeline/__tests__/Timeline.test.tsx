import { TimelineItemProps, Timeline } from '..';
import { render } from '@testing-library/react';
import { testAttrs } from '~/testAttrs';

describe('Timeline', () => {
  testAttrs(Timeline);

  it('basic', () => {
    const items: TimelineItemProps[] = [
      {
        content: 'content 1',
        time: '2023-01-01',
        title: 'title 1',
      },
      {
        content: 'content 2',
        time: '2023-01-02',
        lineType: 'dotted',
        title: 'title 2',
        type: 'error',
      },
    ];
    expect(
      render(<Timeline items={items} />).container.firstChild,
    ).toMatchSnapshot();
  });
});
