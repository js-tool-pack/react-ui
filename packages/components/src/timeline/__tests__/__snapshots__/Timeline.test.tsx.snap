// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline basic 1`] = `
<div
  class="t-timeline t-timeline--left"
>
  <div
    class="t-timeline-item t-timeline-item--default"
    style="--t-timeline-line-style: solid;"
  >
    <div
      class="t-timeline-item__content-block"
    >
      <div
        class="t-timeline-item__title"
      >
        title 1
      </div>
      <div
        class="t-timeline-item__content"
      >
        content 1
      </div>
      <div
        class="t-timeline-item__time"
      >
        2023-01-01
      </div>
    </div>
    <div
      class="t-timeline-item__line-block"
    >
      <div
        class="t-timeline-item__circle"
      />
      <div
        class="t-timeline-item__line"
      />
    </div>
  </div>
  <div
    class="t-timeline-item t-timeline-item--error"
    style="--t-timeline-line-style: dotted;"
  >
    <div
      class="t-timeline-item__content-block"
    >
      <div
        class="t-timeline-item__title"
      >
        title 2
      </div>
      <div
        class="t-timeline-item__content"
      >
        content 2
      </div>
      <div
        class="t-timeline-item__time"
      >
        2023-01-02
      </div>
    </div>
    <div
      class="t-timeline-item__line-block"
    >
      <div
        class="t-timeline-item__circle"
      />
      <div
        class="t-timeline-item__line"
      />
    </div>
  </div>
</div>
`;
